### ISSUE TITLE
Keyboard focus ring is activated when not using the keyboard

### ISSUE THREAD

**Issue by seanh** [hypothesis/lms#2295]
The keyboard focus ring (black border) is sometimes shown around buttons even though I haven't used the keyboard:

![Screenshot from 2020-12-11 13-38-33](https://user-images.githubusercontent.com/22498/101910035-528d2300-3bb6-11eb-90dd-11ca56b74ca7.png)

1. Delete all the access tokens from your DB, to force re-authorization:

   ```terminal
   tox -qe dockercompose -- exec postgres psql -U postgres -c 'DELETE FROM oauth2_token;'
   ```

2. Launch the [localhost (make devdata) Canvas Files Assignment](https://hypothesis.instructure.com/courses/125/assignments/875)

3. You should see the **Authorize Hypothesis** dialog and, even though you haven't used the keyboard, the focus ring black border will be shown around the <kbd>Authorize</kbd> button. Note that the focus ring is difficult to notice because it's black and the button is also black (that's a separate issue), but if you hover the mouse pointer over the <kbd>Authorize</kbd> button the button will turn red and the black focus ring will be revealed.

I _think_ there are also other situations where the focus ring is being shown even though the keyboard isn't being used. The one particular situation described above is just the one way I've found to reproduce it.

**Comment by robertknight** [comment]
It looks like the `Button` component's styles don't use the `outline-on-keyboard-focus` mixin as they do in the client. What we probably need to do here is sync up the LMS frontend to use the current version of the button styles from the client.

**Comment by seanh** [comment]
Another example of this: https://github.com/hypothesis/lms/pull/2329#issuecomment-770751830

I think I've seen other places in the UI where the focus ring is around a dialog as soon as the dialog appears, as well

**Comment by LMS007** [comment]
Closing this out as all dialog buttons have been converted to the new shared button component.

Focus will be visible if the browser determines if the keyboard was used to trigger the action that opened the dialog, otherwise no focus will be present on buttons.
